### ISSUE TITLE
atom-beautify is installed on Windows only if Atom is run as administrator

### ISSUE THREAD

**Issue by dessant** [Glavin001/atom-beautify#902]
I have opened an issue about this before, but it got closed as a duplicate (https://github.com/Glavin001/atom-beautify/issues/681). I have gone through the suggestions I could find, the .atom folder is owned by the current user, none of them worked.

I can only install the package if Atom is run as administrator, other plugins which use spawn-sync do not show this issue.

```
Installing “atom-beautify@0.29.1” failed.Hide output…

> spawn-sync@1.0.15 postinstall C:\Users\prc\AppData\Local\Temp\apm-install-dir-11637-5264-idvrvg\node_modules\atom-beautify\node_modules\cross-spawn\node_modules\spawn-sync
> node postinstall


'node' is not recognized as an internal or external command,
operable program or batch file.
npm WARN optional dep failed, continuing fsevents@1.0.11
npm WARN engine deep-extend@0.4.1: wanted: {"node":">=0.12.0","iojs":">=1.0.0"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine is-buffer@1.1.3: wanted: {"node":">=0.12"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine is-buffer@1.1.3: wanted: {"node":">=0.12"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine is-buffer@1.1.3: wanted: {"node":">=0.12"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine is-buffer@1.1.3: wanted: {"node":">=0.12"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine is-buffer@1.1.3: wanted: {"node":">=0.12"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine is-buffer@1.1.3: wanted: {"node":">=0.12"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine is-buffer@1.1.3: wanted: {"node":">=0.12"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm ERR! Windows_NT 6.2.9200
npm ERR! argv "C:\\Users\\prc\\AppData\\Local\\atom\\app-1.6.2\\resources\\app\\apm\\bin\\node.exe" "C:\\Users\\prc\\AppData\\Local\\atom\\app-1.6.2\\resources\\app\\apm\\node_modules\\npm\\bin\\npm-cli.js" "--globalconfig" "C:\\Users\\prc\\.atom\\.apm\\.apmrc" "--userconfig" "C:\\Users\\prc\\.atom\\.apmrc" "install" "C:\\Users\\prc\\AppData\\Local\\Temp\\d-11637-5264-pxlkkl\\package.tgz" "--target=0.34.5" "--arch=ia32"
npm ERR! node v0.10.40
npm ERR! npm  v2.13.3
npm ERR! code ELIFECYCLE

npm ERR! spawn-sync@1.0.15 postinstall: `node postinstall`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the spawn-sync@1.0.15 postinstall script 'node postinstall'.
npm ERR! This is most likely a problem with the spawn-sync package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node postinstall
npm ERR! You can get their info via:
npm ERR!     npm owner ls spawn-sync
npm ERR! There is likely additional logging output above.
```

**Comment by Glavin001** [comment]
I recommend that you contact the Atom team to help you get Atom setup properly with node.
Here is their current available documentation for common errors for Windows: https://github.com/atom/atom/blob/6f1c1fd0e0242b096f4781698066788b26af2fa4/docs/build-instructions/windows.md#common-errors

**Comment by dessant** [comment]
Shouldn't installing node be unnecessary? I've used the official installer.

**Comment by Glavin001** [comment]
I do not use Windows for developer because of Windows specific problems like this, and so I am unable to help with this issue.
In fact, the `cross-spawn` package was added to fix another Windows bug with Node's spawn method, however its installation is appearing to cause more problems than it solves.

In the next few months I will be implementing new features and removing the spawn-sync package, since it is causing so many problems.

**Comment by Glavin001** [comment]
> Shouldn't installing node be unnecessary?

Yeah, Node should be installed with Atom already. It appears that the node that is installed already is not being found. Hopefully the Atom and/or apm team would know why this happens on Windows.

**Comment by Glavin001** [comment]
I just published a patch in release v0.29.4 that makes `cross-spawn` dependency optional. 
The [`cross-spawn`](https://github.com/IndigoUnited/node-cross-spawn) package depends on [`spawn-sync`](https://github.com/ForbesLindesay/spawn-sync) and has been causing problems when installing Atom-Beautify for many users. By making this dependency optional, users will be able to easily install Atom-Beautify and, in the case that [`cross-spawn`](https://github.com/IndigoUnited/node-cross-spawn) fails to install, will fallback to the builtin [`spawn`](https://nodejs.org/api/child_process.html#child_process_child_process_spawn_command_args_options) method.

**Comment by dessant** [comment]
I've uninstalled the plugin and tried installing 0.29.7 (from the settings panel), but it fails and the fallback option does not seem to be used:

```
> spawn-sync@1.0.15 postinstall C:\Users\prc\AppData\Local\Temp\apm-install-dir-116327-1660-1myqrb3\node_modules\atom-beautify\node_modules\cross-spawn\node_modules\spawn-sync
> node postinstall


> atom-beautify@0.29.7 postinstall C:\Users\prc\AppData\Local\Temp\apm-install-dir-116327-1660-1myqrb3\node_modules\atom-beautify
> node script/build-options.js


npm WARN engine analytics-node@2.1.0: wanted: {"node":">= 0.12.x"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine typescript-formatter@2.1.0: wanted: {"node":">= 4.2.0"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN deprecated lodash@1.2.1: lodash@<3.0.0 is no longer maintained. Upgrade to lodash@^4.0.0.
'node' is not recognized as an internal or external command,
operable program or batch file.
npm WARN optional dep failed, continuing cross-spawn@2.2.3
npm WARN optional dep failed, continuing fsevents@1.0.11
npm WARN engine deep-extend@0.4.1: wanted: {"node":">=0.12.0","iojs":">=1.0.0"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine is-buffer@1.1.3: wanted: {"node":">=0.12"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine is-buffer@1.1.3: wanted: {"node":">=0.12"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine is-buffer@1.1.3: wanted: {"node":">=0.12"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine is-buffer@1.1.3: wanted: {"node":">=0.12"} (current: {"node":"0.10.40","npm":"2.13.3"})
npm WARN engine is-buffer@1.1.3: wanted: {"node":">=0.12"} (current: {"node":"0.10.40","npm":"2.13.3"})
'node' is not recognized as an internal or external command,
operable program or batch file.
npm ERR! Windows_NT 6.2.9200
npm ERR! argv "C:\\Users\\prc\\AppData\\Local\\atom\\app-1.7.2\\resources\\app\\apm\\bin\\node.exe" "C:\\Users\\prc\\AppData\\Local\\atom\\app-1.7.2\\resources\\app\\apm\\node_modules\\npm\\bin\\npm-cli.js" "--globalconfig" "C:\\Users\\prc\\.atom\\.apm\\.apmrc" "--userconfig" "C:\\Users\\prc\\.atom\\.apmrc" "install" "C:\\Users\\prc\\AppData\\Local\\Temp\\d-116327-1660-143ni4c\\package.tgz" "--target=0.36.8" "--arch=ia32"
npm ERR! node v0.10.40
npm ERR! npm  v2.13.3
npm ERR! code ELIFECYCLE

npm ERR! atom-beautify@0.29.7 postinstall: `node script/build-options.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the atom-beautify@0.29.7 postinstall script 'node script/build-options.js'.
npm ERR! This is most likely a problem with the atom-beautify package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node script/build-options.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls atom-beautify
npm ERR! There is likely additional logging output above.

```
